### PR TITLE
Remove `new` keyword in a few files

### LINKED ISSUE #104436: There is still code with the old `new` keyword
Use the following settings in VS Code and search in all files
image

Not all results contain code with the old `new` keyword. But there are some files.

### PR DESCRIPTION
**Pull request by nilsreichardt**
There were files with the old keyword.

> we really should find a way to analyze this

Wrote @Hixie in https://github.com/flutter/flutter/pull/69507/files#r519475809

Therefore, I tried to find all old `new` keywords. I think I fixed them all. But there are still some that I wasn't sure were on purpose there:
* analyze_snippet_code_test.dart L28, L30
* known_broken_documentation.dart L30, 62
* assertions_test.dart L356, L360, L391, L395, L396
* error_reporting_test.dart L227, L249
* stack_frame_test.dart L107, L123, L140, L198
* expression_evaluation_test.dart L160, L166

Fixes #104436

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/painting/text_style.dart [hunk 1]
```diff
@@ -14,10 +14,10 @@ import 'text_painter.dart';
 const String _kDefaultDebugLabel = 'unknown';
 
 const String _kColorForegroundWarning = 'Cannot provide both a color and a foreground\n'
-    'The color argument is just a shorthand for "foreground: new Paint()..color = color".';
+    'The color argument is just a shorthand for "foreground: Paint()..color = color".';
 
 const String _kColorBackgroundWarning = 'Cannot provide both a backgroundColor and a background\n'
-    'The backgroundColor argument is just a shorthand for "background: new Paint()..color = color".';
+    'The backgroundColor argument is just a shorthand for "background: Paint()..color = color".';
```
Comment (nilsreichardt) [inline packages/flutter/lib/src/painting/text_style.dart:20]: Should I add tests for this?

**Comment by goderbauer** [comment]
Ideally, we'd wrap the code samples in `{@tool snippet}` sections so they would get analyzed...

**Review by goderbauer — APPROVED** [review]
LGTM

Thanks for the clean-up.

**Review by Piinks — APPROVED** [review]
LGTM!

**Comment by nilsreichardt** [comment]
What is with the files in the description?

> analyze_snippet_code_test.dart L28, L30
known_broken_documentation.dart L30, 62
assertions_test.dart L356, L360, L391, L395, L396
error_reporting_test.dart L227, L249
stack_frame_test.dart L107, L123, L140, L198
expression_evaluation_test.dart L160, L166

Should I open new a PR and you just say if removing the `new` was not correct?

**Comment by Piinks** [comment]
Those look fine to me, I do not think they need to be changed. Thanks!

**Pull request merged**